src/main/java/io/axoniq/axonserver/connector/event/impl/BufferedAggregateEventStream.java

```diff
@@ -98,4 +105,20 @@ protected GetAggregateEventsRequest buildFlowControlMessage(FlowControl flowCont
     protected Event terminalMessage() {
         return TERMINAL_MESSAGE;
     }
+
+    @Override
+    public void onNext(Event event) {
+        Event prevEvent = lastReceivedEvent.get();
+        if (prevEvent == null || prevEvent.getAggregateSequenceNumber() + 1 == event.getAggregateSequenceNumber()) {
+            super.onNext(event);
+            lastReceivedEvent.set(event);
+        } else {
+            String message = String.format("Invalid sequence number for aggregate %s. Received: %d, expected: %d",
+                                           event.getAggregateIdentifier(),
+                                           event.getAggregateSequenceNumber(),
+                                           prevEvent.getAggregateSequenceNumber() + 1);
+            logger.error(message);
+            this.onError(new RuntimeException(message));
```

Comment:
This will report an error toward the client, but I don't believe it will stop the server necessarily from communicating any further updates. Shouldn't we close the call somehow?

Reply:
Yes, we should signal `onError`.